server/mediaserver.go

```diff
@@ -357,12 +353,46 @@ func gotRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.
 					s.LivepeerNode.Database.SetSegmentCount(jobId, int64(seg.SeqNo))
 				}
 
-				s.LivepeerNode.VideoSource.InsertHLSSegment(hlsStrmID, seg)
+				seg.Name = "" // hijack seg.Name to convey the uploaded URI
+				name := fmt.Sprintf("%s/%d.ts", vProfile.Name, seg.SeqNo)
+				uri, err := cpl.GetOSSession().SaveData(name, seg.Data)
+				if err != nil {
+					glog.Errorf("Error saving segment %d: %v", seg.SeqNo, err)
+					if monitor.Enabled {
+						monitor.LogSegmentUploadFailed(nonce, seg.SeqNo, err.Error())
+					}
+					return
+				}
+				if drivers.NodeStorage.IsExternal() {
```

Comment:
isn't this check assumes that `PlaylistManager` (`cpl`) has OS session created from `drivers.NodeStorage`?

Reply:
PlaylistManager will always have its OS session created from `drivers.NodeStorage` . At least, right now.

Agreed that it's not super future-proof. We could have the IsExternal check be a part of `OSSession` but we'd still need to either retain the OSDriver check, or create+discard a session within RPC `getTranscoder`

Reply:
Fixed and pushed, but untested; wanted to get it out for review.